test/test-quora-duplicate-question-pair.rb

```diff
@@ -8,26 +8,13 @@ def record(*args)
   end
 
   test("#each") do
-    records = @dataset.each.to_a
-    assert_equal([
-                   404290,
-                   record(0,
-                          1,
-                          2,
-                          "What is the step by step guide to invest in share market in india?",
-                          "What is the step by step guide to invest in share market?",
-                          false),
-                   record(404289,
-                          537932,
-                          537933,
-                          "What is like to have sex with cousin?",
-                          "What is it like to have sex with your cousin?",
-                          false),
-                 ],
-                 [
-                   records.size,
-                   records.first,
-                   records.last,
-                 ])
+    records = @dataset
+    assert_equal(record(0,
+                        1,
+                        2,
+                        "What is the step by step guide to invest in share market in india?",
+                        "What is the step by step guide to invest in share market?",
+                        false),
+                 records.first)
```

Comment:
Could you use `#each` because this is a test for `#each`?

```suggestion
    assert_equal(record(0,
                        1,
                        2,
                        "What is the step by step guide to invest in share market in india?",
                        "What is the step by step guide to invest in share market?",
                        false),
                 @dataset.each.next)
```

Reply:
Thanks! Certainly this is a test for `#each`.